process/block/gasConsumption.go

```diff
@@ -0,0 +1,158 @@
+package block
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/sharding"
+)
+
+const (
+	initialLimitsFactor = uint64(100)
+	// TODO: move these to config
+	percentSplitBlock         = uint64(50) // 50%
+	percentDecreaseLimitsStep = uint64(10) // 10%
+)
+
+type gasConsumption struct {
+	sync.RWMutex
+	economicsFee            process.FeeHandler
+	shardCoordinator        sharding.Coordinator
+	totalGasConsumedInShard map[uint32]uint64
+	gasConsumedByMiniBlock  map[string]uint64
```

Comment:
could we end up having a huge map here? or some mechanism to cleanup old miniblocks, instead of just `Reset` all?

Reply:
not needed, this component should be reset for each block